Source/CombatExtended/CombatExtended/Verbs/Verb_MeleeAttackCE.cs

```diff
@@ -227,8 +227,17 @@ public override bool TryCastShot()
                 MoteMakerCE.ThrowText(targetThing.PositionHeld.ToVector3Shifted(), targetThing.MapHeld, moteText);
             }
             soundDef.PlayOneShot(new TargetInfo(targetThing.PositionHeld, targetThing.MapHeld));
-            casterPawn.Drawer.Notify_MeleeAttackOn(targetThing);
-            if (defender != null && !defender.Dead)
```

Comment:
You are removing the `!defender.Dead`.  I would expect the sane check to be `(defender != null && !defender.Dead && defender.Spawned)`